cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -212,6 +224,63 @@ func (s *Service) Delete(ctx context.Context, spec interface{}) error {
 	return err
 }
 
+func (s *Service) getAddresses(ctx context.Context, vm compute.VirtualMachine) ([]corev1.NodeAddress, error) {
+
+	addresses := []corev1.NodeAddress{}
+
+	for _, nicRef := range *vm.NetworkProfile.NetworkInterfaces {
+
+		// The full ID includes the name at the very end. Split the string and pull the last element
+		// Ex: /subscriptions/$SUB/resourceGroups/$RG/providers/Microsoft.Network/networkInterfaces/$NICNAME
```

Comment:
What if it doesnt? What if ID is empty?

Can you please check for empty or atleast make sure that SpltString is valid?